src/caffe/parallel.cpp

```diff
@@ -318,6 +318,15 @@ void P2PSync<Dtype>::on_start() {
     CUDA_CHECK(cudaStreamSynchronize(cudaStreamDefault));
     children_[i]->queue_.push(this);
   }
+
+  // Move syncedmem head to HEAD_AT_GPU
+  const vector<Blob<Dtype>*>& params = solver_->net()->learnable_params();
+  Dtype* mutable_gpu_data;
+  for (int i = 0; i < params.size(); ++i) {
+    mutable_gpu_data = params[i]->mutable_gpu_data();
+    mutable_gpu_data++; // added to prevent the warning message of compilers
```

Comment:
Why do you need to hold onto the return value?
You could just call the `mutable_gpu_data` function:
```
  for (int i = 0; i < params.size(); ++i) {
    params[i]->mutable_gpu_data();
  }
````